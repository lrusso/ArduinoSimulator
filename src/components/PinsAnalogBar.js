import React from "react"
import PinsAnalogLabel from "./PinsAnalogLabel"
import PinsAnalogItem from "./PinsAnalogItem"
import { useSimulatorContext } from "../contexts/SimulatorContext"
import { BOARD_MEGA1280, BOARD_MEGA2560, BOARD_NANO } from "../utils/service"

const PinsAnalogBar = () => {
  const {
    analogPin0,
    analogPin1,
    analogPin2,
    analogPin3,
    analogPin4,
    analogPin5,
    analogPin6,
    analogPin7,
    analogPin8,
    analogPin9,
    analogPin10,
    analogPin11,
    analogPin12,
    analogPin13,
    analogPin14,
    boardType,
  } = useSimulatorContext()

  const isMega = boardType === BOARD_MEGA1280 || boardType === BOARD_MEGA2560
  const showPin6And7 = boardType === BOARD_NANO || isMega

  return (
    <div style={styles.container}>
      <div style={styles.noScrollbar}>
        <div style={{ width: isMega ? "850px" : "550px", ...styles.wrapper }}>
          <PinsAnalogLabel />
          <PinsAnalogItem value={analogPin0} />
          <PinsAnalogItem value={analogPin1} />
          <PinsAnalogItem value={analogPin2} />
          <PinsAnalogItem value={analogPin3} />
          <PinsAnalogItem value={analogPin4} />
          <PinsAnalogItem value={analogPin5} />
          {showPin6And7 && (
            <>
              <PinsAnalogItem value={analogPin6} />
              <PinsAnalogItem value={analogPin7} />
            </>
          )}
          {isMega && (
            <>
              <PinsAnalogItem value={analogPin8} />
              <PinsAnalogItem value={analogPin9} />
              <PinsAnalogItem value={analogPin10} />
              <PinsAnalogItem value={analogPin11} />
              <PinsAnalogItem value={analogPin12} />
              <PinsAnalogItem value={analogPin13} />
              <PinsAnalogItem value={analogPin14} />
            </>
          )}
        </div>
      </div>
    </div>
  )
}

const styles = {
  container: {
    height: "26px",
    overflowY: "hidden",
  },
  noScrollbar: {
    overflowX: "scroll",
    overflowY: "hidden",
    outline: "none",
    height: "80px",
  },
  wrapper: {
    float: "left",
  },
}

export default PinsAnalogBar
