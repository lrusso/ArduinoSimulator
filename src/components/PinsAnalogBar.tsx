import React from "react"
import PinsAnalogLabel from "./PinsAnalogLabel"
import PinsAnalogItem from "./PinsAnalogItem"
import { useSimulatorContext } from "../contexts/SimulatorContext"
import { isMega, isNano } from "../../src/utils/service"

const PinsAnalogBar = () => {
  const { analogPins, boardType } = useSimulatorContext()

  const isMegaBoard = isMega(boardType)
  const isNanoBoard = isNano(boardType)

  return (
    <div style={styles.container}>
      <div style={styles.noScrollbar}>
        <div style={{ width: isMegaBoard ? "850px" : "550px", ...styles.wrapper }}>
          <PinsAnalogLabel />
          <div className="flex-row">
          {analogPins.slice(0, 6).map((pin, index) => (
            <PinsAnalogItem key={index} gpioAnalog={pin} />
          ))}
          {(isMegaBoard || isNanoBoard) && (
            <>
              {analogPins.slice(6, 8).map((pin, index) => (
                <PinsAnalogItem key={index} gpioAnalog={pin} />
              ))}
            </>
          )}
          {isMegaBoard && (
            <>
              {analogPins.slice(8, 14).map((pin, index) => (
                <PinsAnalogItem key={index} gpioAnalog={pin} />
              ))}
            </>
          )}
          </div>
        </div>
      </div>
    </div>
  )
}

const styles: { [key: string]: React.CSSProperties } = {
  container: {
    height: "36px",
    overflowY: "hidden",
    borderBottom: "1px solid #363636",
    marginBottom: "5px"
  },
  noScrollbar: {
    overflowX: "scroll",
    overflowY: "hidden",
    outline: "none",
    height: "80px",
  },
  wrapper: {
    display: "flex",
    flexBasis: "0",
    flexDirection: "row",
    alignItems: "baseline",
    gap: "2px",
    flexWrap: "nowrap",
  }
}

export default PinsAnalogBar
