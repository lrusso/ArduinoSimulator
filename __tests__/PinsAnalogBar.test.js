import React from "react"
import renderer from "react-test-renderer"
import PinsAnalogBar from "../src/components/PinsAnalogBar"

describe("PinsAnalogBar", () => {
  test("Rendering 1", () => {
    const tree = renderer.create(<PinsAnalogBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
