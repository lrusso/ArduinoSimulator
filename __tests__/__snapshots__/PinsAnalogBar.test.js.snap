// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinsAnalogBar Rendering 1 1`] = `
<div
  style={
    {
      "height": "26px",
      "overflowY": "hidden",
    }
  }
>
  <div
    style={
      {
        "height": "80px",
        "outline": "none",
        "overflowX": "scroll",
        "overflowY": "hidden",
      }
    }
  >
    <div
      style={
        {
          "float": "left",
          "width": "550px",
        }
      }
    >
      <div
        style={
          {
            "cursor": "default",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "13px",
            "fontWeight": "bold",
            "lineHeight": 2,
            "marginLeft": "7px",
            "marginRight": "1px",
          }
        }
      >
        ANALOG PINS
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "40px",
          }
        }
      />
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "40px",
          }
        }
      />
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "40px",
          }
        }
      />
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "40px",
          }
        }
      />
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "40px",
          }
        }
      />
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "40px",
          }
        }
      />
    </div>
  </div>
</div>
`;
